components/servo/lib.rs

```diff
@@ -217,6 +217,21 @@ pub struct InitializedServo<Window: WindowMethods + 'static + ?Sized> {
     pub browser_id: TopLevelBrowsingContextId,
 }
 
+/// Utility to help forward embedder event to constellation.
+macro_rules! forward_to_constellation {
+    ($self:ident, $variant:ident $(($($rest:tt),*))?) => {
+        {
+            let msg = ConstellationMsg::$variant$(($($rest),*))?;
+            if let Err(e) = $self.constellation_chan.send(msg) {
+                warn!(
+                    "Sending {} message to constellation failed ({:?}).",
+                    stringify!($variant), e,
+                );
+            }
+        }
+    };
+}
```

Comment:
I think this will be easier to read as a function, if possible and also friendlier for language servers in the IDE. Just make sure that the message supports `Debug`.

Reply:
done in 5364e71b8372de9e2c01ed3b8e5c81f59c93efe8